rancher/dashboard
# Added cluster tool link in cluster group nav section

### Summary
Fixes #9228

Relocate the Cluster Tools button

Note: The `ifHaveType` mechanism did not support multiple types - this has been added in this PR, so that we show the Tools item in the sidenav only when user has access to the required types.

### How to test
Ref XD link: https://xd.adobe.com/view/30c5a51c-e7b7-4ebf-b75f-1dc37391d3d7-6596/

- Go to local cluster => side nav should have `Tools` nav link under `Cluster` group
- `Cluster Tools` Should not be there at the bottom of side nav
-  On the cluster dashboard page > `Install Monitoring` text has changed to `Cluster Tools` and 
`Add Cluster Badge` link has changed to an icon button with a tooltip 'Customize Appearance'

### Manual Tests

- Check that the side nav renders correctly
- Check the icon button opens the customize appearance dialog
- Check that clicking on 'Tools' in the side nav takes you to cluster tools
- Check that clicking on 'Cluster Tools' on the dashboard page takes you to the cluster tools page
- Install a cluster tool like monitoring and ensure that the side nav updates to show 'Monitoring'


### Screenshot/Video

![image](https://github.com/rancher/dashboard/assets/1955897/1bb2ca99-7d7c-4b86-9900-95fe34d87854)
